File: trakt-api/src/main/java/net/simonvt/cathode/api/entity/Movie.java
/*
 * Copyright (C) 2014 Simon Vig Therkildsen
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package net.simonvt.cathode.api.entity;

import java.util.List;

public class Movie {

  String title;

  Integer year;

  Ids ids;

  String tagline;

  String overview;

  String released;

  Integer runtime;

  String updatedAt;

  String trailer;

  String homepage;

  Float rating;

  String language;

  List<String> availableTranslations;

  Images images;

  List<String> genres;

  String certification;

  public String getTitle() {
    return title;
  }

  public Integer getYear() {
    return year;
  }

  public Ids getIds() {
    return ids;
  }

  public String getTagline() {
    return tagline;
  }

  public String getOverview() {
    return overview;
  }

  public String getReleased() {
    return released;
  }

  public Integer getRuntime() {
    return runtime;
  }

  public String getUpdatedAt() {
    return updatedAt;
  }

  public String getTrailer() {
    return trailer;
  }

  public String getHomepage() {
    return homepage;
  }

  public Float getRating() {
    return rating;
  }

  public String getLanguage() {
    return language;
  }

  public List<String> getAvailableTranslations() {
    return availableTranslations;
  }

  public Images getImages() {
    return images;
  }

  public List<String> getGenres() {
    return genres;
  }

  public String getCertification() {
    return certification;
  }
}

File: trakt-api/src/main/java/net/simonvt/cathode/api/entity/Movie.kt
/*
 * Copyright (C) 2014 Simon Vig Therkildsen
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package net.simonvt.cathode.api.entity

data class Movie(
  val title: String? = null,
  val year: Int? = null,
  val ids: Ids,
  val tagline: String? = null,
  val overview: String? = null,
  val released: String? = null,
  val runtime: Int? = null,
  val updated_at: String? = null,
  val trailer: String? = null,
  val homepage: String? = null,
  val rating: Float? = null,
  val votes: Int? = null,
  val language: String? = null,
  val available_translations: List<String>? = null,
  val genres: List<String>? = null,
  val certification: String? = null
)
